File: queue/python/Queue.py
class Queue:
    def __init__(self):
        self.queue = []
        self.size = 0

    def push(self, element):
        self.queue.append(element)
        self.size = 1

    def pop(self):
        if not self.isEmpty():
            self.queue.pop(0)
            self.size -= 1
        return "Queue is Empty!!!"

    def front(self):
        if not self.isEmpty():
            return self.queue[0]
        return "Queue is Empty!!!"

    def length(self):
        return self.size

    def isEmpty(self):
        return len(self.queue) == 0


q = Queue()
print(q.isEmpty())
q.push(1)
q.push(2)
q.push(3)
q.push(4)
print(q.isEmpty())
print(q.front())
q.pop()
q.pop()
q.pop()
q.pop()
print(q.isEmpty())
print(q.front())

File: queue/c/Queue.c
#include <limits.h>
#include <stdio.h>
#include <stdlib.h>

struct Queue
{
  int front, rear, size;
  int *array;
  unsigned capacity;
};

struct Queue *createQueue(unsigned capacity)
{
  struct Queue *queue = (struct Queue *)malloc(sizeof(struct Queue));
  queue->capacity = capacity;
  queue->front = queue->size = 0;
  queue->rear = capacity - 1;
  queue->array = (int *)malloc(sizeof(queue->capacity * sizeof(int)));
  return queue;
}

int isFull(struct Queue *queue)
{
  return queue->size == queue->capacity;
}

int isEmpty(struct Queue *queue)
{
  return queue->size == 0;
}

void push(struct Queue *queue, int element)
{
  if (isFull(queue))
  {
    return;
  }
  queue->rear = (queue->rear + 1) % queue->capacity;
  queue->array[queue->rear] = element;
  queue->size = queue->size + 1;
  printf("%d pushed to queue\n", element);
}

int pop(struct Queue *queue)
{
  if (isEmpty(queue))
  {
    return INT_MIN;
  }
  int item = queue->array[queue->front];
  queue->front = (queue->front + 1) % queue->capacity;
  queue->size = queue->size - 1;
  return item;
}

int front(struct Queue *queue)
{
  if (isEmpty(queue))
  {
    return INT_MIN;
  }
  return queue->array[queue->front];
}

int rear(struct Queue *queue)
{
  if (isEmpty(queue))
  {
    return INT_MIN;
  }
  return queue->array[queue->rear];
}

void length(struct Queue *queue)
{
  printf("Elements on stack: %d\n", queue->front + 1);
}

int main()
{
  struct Queue *queue = createQueue(1000);

  push(queue, 10);
  push(queue, 20);
  push(queue, 30);
  push(queue, 40);

  printf("%d dequeued from queue\n\n", pop(queue));

  printf("Front item is %d\n", front(queue));
  printf("Rear item is %d\n", rear(queue));

  return 0;
}